### PR TITLE
add test-integration-cli specifics for userns

### PR DESCRIPTION
**Pull request by jessfraz**
Add test-integration-cli vars

so you can see the hang in the ensure-http-server, the daemon logs are in bundles/latest/test-integration-cli/docker.log

you can run with 

```
DOCKER_REMAP_ROOT=default-er-whatever make test-integration-cli
```

### CODE REVIEW

**Comment by jessfraz** [comment]
for fun if you comment out this line you can run the tests https://github.com/docker/docker/blob/master/hack/make/.integration-daemon-setup#L5

**Comment by jessfraz** [comment]
ok cherry pick!!! it works, but doesnt pass but that is good :)

File: hack/make/.integration-daemon-setup [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 bundle .ensure-emptyfs
 bundle .ensure-frozen-images
-bundle .ensure-httpserver
```
Comment (jessfraz) [inline hack/make/.integration-daemon-setup]: no wait dont i forgot i had this

Comment (jessfraz) [inline hack/make/.integration-daemon-setup]: hmm i cant get it to work on overlay :/

Comment (estesp) [inline hack/make/.integration-daemon-setup]: Let me spin up my VM with overlay and see what I can figure out

Jess Frazelle wrote:

**Comment by jessfraz** [comment]
so if we get this https://github.com/docker/docker/commit/aef440346ddd901264fe2e601948b5b3d9f2b5a7#commitcomment-13570124

then I can fix the rest of em ;)

**Comment by jessfraz** [comment]
ok the tests all pass when run with `DOCKER_REMAP_ROOT=default`, comments were added to the tests where certain files return permission denied, we can make it contributor issue to try and resolve them to work.... or i can try and fix after i fix windows servers that are freaking out :P

**Comment by estesp** [comment]
uh oh, is there a check inversion here in the `requirements`? Notice after push and jenkins restart even the "normal" janky run has a lot of this:

```
DockerSuite.TestRunAddingOptionalDevices (Test cannot be run when remapping root)
```

**Comment by jessfraz** [comment]
shit let me check

**Comment by jessfraz** [comment]
nooooooooo :( i will fix

**Comment by jessfraz** [comment]
this makes no sense it works for me locally

**Comment by jessfraz** [comment]
its a conspiracy, the check needs to be `if len(root) == 0 {`

**Pull request closed without merging**